cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/common/DataprocUtils.java

```diff
@@ -316,16 +329,14 @@ public static void emitMetric(ProvisionerContext context, String region,
         statusCode = StatusCode.Code.INTERNAL;
       }
     }
-    Map<String, String> tags = ImmutableMap.<String, String>builder()
-        .put("reg", region)
-        .put("sc", statusCode.toString())
-        .build();
-    ProvisionerMetrics metrics = context.getMetrics(tags);
-    metrics.count(metricName, 1);
-  }
-
-  public static void emitMetric(ProvisionerContext context, String region, String metricName) {
-    emitMetric(context, region, metricName, null);
+    ImmutableMap.Builder<String, String> tags = ImmutableMap.<String, String>builder()
+        .put("reg", dataprocMetric.getRegion())
+        .put("sc", statusCode.toString());
+    if (dataprocMetric.getLaunchMode() != null) {
+      tags.put("lchmode", dataprocMetric.getLaunchMode().name());
```

Comment:
I don't see any aggregation for this metric in `DefaultMetricsStore`: https://github.com/cdapio/cdap/blob/50621d89331e310a31ab8369aa596ab63227fb59/cdap-watchdog/src/main/java/io/cdap/cdap/metrics/store/DefaultMetricStore.java#L108

IIUC, users won't be able to query this metric using CDAP metrics api.

Reply:
Although extensions will still be able to publish it because they read from TMS.

Reply:
yeah that's correct, it won't be queryable through CDAP. It looked none of the existing dataproc metrics could be (couldn't find 'reg' or 'sc' tags either), so didn't want to introduce a new CDAP aggregation.